src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.csproj

```diff
@@ -54,6 +54,14 @@
     <AssemblyOriginatorKeyFile>..\Common\PublicKey.snk</AssemblyOriginatorKeyFile>
   </PropertyGroup>
   <ItemGroup>
+    <Reference Include="Microsoft.AI.PerfCounterCollector, Version=2.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\packages\Microsoft.ApplicationInsights.PerfCounterCollector.2.2.0\lib\net45\Microsoft.AI.PerfCounterCollector.dll</HintPath>
```

Comment:
Consider including [Microsoft.ApplicationInsights.WindowsApps](https://www.nuget.org/packages/Microsoft.ApplicationInsights.WindowsApps/) and using `Microsoft.ApplicationInsights.WindowsServer.UnhandledExceptionTelemetryModule` and `Microsoft.ApplicationInsights.WindowsServer.UnobservedExceptionTelemetryModule`. 

https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/8c51d66c08c1baf0844613637fa5d3ccfc7590f9/Test/Web/FunctionalTests/TestApps/ConsoleAppFW40/ApplicationInsights.config#L9-L10

Those will handle everything that crashes the process. In future I'd recommend also to subscribe to `FirstChanceExceptionStatisticsTelemetryModule`, but this is only available in beta right now